### PR TITLE
Use private IPs only for CI

### PR DESCRIPTION
**Pull request by timrchavez**
Because the test VMs share a private subnet with the Jenkins server, we
do not need to access them using public IP addresses.

### CODE REVIEW

**Comment by j2sol** [comment]
This is looking good. What more do you want to do before removing the "WIP" flag?

**Comment by timrchavez** [comment]
The --extra-vars is not actually working....

**Comment by timrchavez** [comment]
This works now

File: playbooks/ci-full/tasks/create.yml [hunk 1]
```diff
@@ -48,11 +49,22 @@
         wait: true
       with_items: "{{ testenv_instance_names }}"
       register: testenv_floating_ips
+      when: use_private_ips == false
```
Comment (masteinhauser) [inline playbooks/ci-full/tasks/create.yml]: You want to use the `|bool` filter here to help avoid comparing strings and breaking on falsey values like `no`.

Comment (j2sol) [inline playbooks/ci-full/tasks/create.yml]: Seconded.

File: playbooks/ci-full/tasks/create.yml [hunk 2]
```diff
@@ -48,11 +49,22 @@
...
 
-    - name: wait for instances to boot
+    - name: wait for instances to boot using public ip
       wait_for:
         port: 22
         delay: 5
         timeout: 600
         host: "{{ item.floating_ip.floating_ip_address }}"
       with_items: "{{ testenv_floating_ips.results }}"
+      when: use_private_ips == false
```
Comment (j2sol) [inline playbooks/ci-full/tasks/create.yml]: Bool filter here too

File: playbooks/ci-full/tasks/create.yml [hunk 3]
```diff
@@ -48,11 +49,22 @@
...
+
+    - name: wait for instances to boot using private ip
+      wait_for:
+        port: 22
+        delay: 5
+        timeout: 600
+        host: "{{ item.openstack.addresses.internal.0.addr }}"
+      with_items: "{{ instances.results }}"
+      when: use_private_ips
```
Comment (j2sol) [inline playbooks/ci-full/tasks/create.yml]: Should use the bool filter here too.

**Pull request merged**